### PR TITLE
Pokemon stats from js to ts

### LINKED ISSUE #607: ✨ [Feature request]: Pokemon stats javascript to typescript
**Is your feature request related to a problem? Please describe.**
This feature is not related to any problem its modification of pokemon-stats from js to ts.

**Describe the solution you'd like**
 jsx to tsx  🛠️

I would like to work on this.

### PR DESCRIPTION
**Pull request by yung-coder**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Tried to convert Pokemon-stats from js to ts

Fixes #607 

## Type of change

- [ ✅] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

It has been tested by me in my local system.

# Checklist:

- [ ✅] I have performed a self-review of my own code
- [ ✅] I have commented my code, particularly in hard-to-understand areas
- [ ✅] I have made corresponding changes to the documentation
- [ ✅] My changes generate no new warnings
- [ ✅] I have added tests that prove my fix is effective or that my feature works
- [ ✅] New and existing unit tests pass locally with my changes
- [ ✅] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by yung-coder** [comment]
Hey @atapas  can you review it..

**Comment by atapas** [comment]
@Sachin-chaurasiya Can you please review this one?

**Comment by Sachin-chaurasiya** [comment]
> @Sachin-chaurasiya Can you please review this one?

Sure, I will do that.

**Comment by yung-coder** [comment]
their is also a pending review in activities can you review it too @Sachin-chaurasiya ..?

File: src/plays/pokemon-stats/components/Pokemoncard.tsx [hunk 1]
```diff
@@ -1,6 +1,42 @@
...
+}
+
+export interface Stat2 {
+  name: string;
+}
+
+export interface Type {
+  type: Type2;
+}
+
```
Comment (Sachin-chaurasiya) [inline src/plays/pokemon-stats/components/Pokemoncard.tsx:23]: Can we create a separate file for interfaces?

File: src/plays/pokemon-stats/components/Search.tsx [hunk 2]
```diff
@@ -1,6 +1,22 @@
 import React from "react";
 import "./search.css";
-const search = (props) => {
+type Propsi = {
+  setname: React.Dispatch<React.SetStateAction<string>>;
```
Comment (Sachin-chaurasiya) [inline src/plays/pokemon-stats/components/Search.tsx:4]: @yung-coder , We should never pass the state action as a prop. we can create a handler like this and pass it as props.

```js
const handleSetName=(value:valueType)=>{
// set the state here 
}
```
Also change the `setname` prop to `onSetName`.

File: src/plays/pokemon-stats/components/Search.tsx [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
+  name: string;
+  fetchPokemonDetails: (e: React.FormEvent<HTMLFormElement>) => void;
+};
+
+type Props = React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  type?: "submit";
+};
```
Comment (Sachin-chaurasiya) [inline src/plays/pokemon-stats/components/Search.tsx:11]: @yung-coder , We can use the interface here and extend the `ButtonHTMLAttributes`.
Also, we don't need to define the `type` explicitly, `ButtonHTMLAttribute` already provide the type like this.
```
type?: 'submit' | 'reset' | 'button' | undefined;
```
You can create your prop like this.
```ts
interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {}
```

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]

**Comment by Sachin-chaurasiya** [comment]
> their is also a pending review in activities can you review it too @Sachin-chaurasiya ..?

@yung-coder , We will do that, please have some patience.

File: src/plays/pokemon-stats/components/Pokemoncard.tsx (hunk 1 above)
Comment (yung-coder) [reply src/plays/pokemon-stats/components/Pokemoncard.tsx:23]: Sure i will .

File: src/plays/pokemon-stats/PokemonStats.tsx [hunk 4]
```diff
@@ -6,12 +6,15 @@ import Pokemoncard from "./components/Pokemoncard";
...
+
+
 // WARNING: Do not change the entry componenet name
-function PokemonStats(props) {
+function PokemonStats(props: any) {
   // Your Code Start below.
   const [pokemon, setpokemon] = useState();
-  const [name, setname] = useState("");
-  const fetchPokemonDetails = (e) => {
+  const [name, setname] = useState<string>("");
```
[the remark this answers is not in the document]
Comment (yung-coder) [reply src/plays/pokemon-stats/PokemonStats.tsx:16]: done

**Comment by yung-coder** [comment]
hey @Sachin-chaurasiya  made the changes please give it a look

File: src/plays/pokemon-stats/PokemonStats.tsx [hunk 5]
```diff
@@ -6,12 +6,15 @@ import Pokemoncard from "./components/Pokemoncard";
 import Search from "./components/Search";
 import { ToastContainer, toast } from "react-toastify";
 import "react-toastify/dist/ReactToastify.css";
+import React from 'react';
+
+
 // WARNING: Do not change the entry componenet name
-function PokemonStats(props) {
+function PokemonStats(props: any) {
```
[the remark this answers is not in the document]
Comment (Sachin-chaurasiya) [reply src/plays/pokemon-stats/PokemonStats.tsx:13]: @yung-coder , If we are doing something then we should do it properly, I suggest you explore more and add the proper prop type.

File: src/plays/pokemon-stats/PokemonStats.tsx [hunk 6]
```diff
@@ -6,15 +6,21 @@ import Pokemoncard from "./components/Pokemoncard";
...
   // Your Code Start below.
-  const [pokemon, setpokemon] = useState();
-  const [name, setname] = useState("");
-  const fetchPokemonDetails = (e) => {
+  const [pokemon, setpokemon] = useState<Root>();
+  const [Name, onSetName] = useState<string>("");
```
Comment (Sachin-chaurasiya) [inline src/plays/pokemon-stats/PokemonStats.tsx:16]:
```suggestion
  const [name, onSetName] = useState<string>("");
```

File: src/plays/pokemon-stats/components/Search.tsx [hunk 7]
```diff
@@ -1,31 +1,37 @@
 import React from "react";
 import "./search.css";
-const search = (props) => {
-  const { setname, fetchPokemonDetails, name } = props;
+type SearchProps = {
+  eventchange:(e: React.ChangeEvent<HTMLInputElement>) => void;
```
Comment (Sachin-chaurasiya) [inline src/plays/pokemon-stats/components/Search.tsx:4]:
```suggestion
  onEventChange:(e: React.ChangeEvent<HTMLInputElement>) => void;
```

File: src/plays/pokemon-stats/components/Search.tsx [hunk 8]
```diff
@@ -1,31 +1,37 @@
...
+  fetchPokemonDetails: (e: React.FormEvent<HTMLFormElement>) => void;
+  Name: string;
```
Comment (Sachin-chaurasiya) [inline src/plays/pokemon-stats/components/Search.tsx:6]:
```suggestion
  name: string;
```

File: src/plays/pokemon-stats/components/Search.tsx [hunk 9]
```diff
@@ -1,31 +1,37 @@
...
+};
+interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {}
+
+const search = (props: SearchProps) => {
```
Comment (Sachin-chaurasiya) [inline src/plays/pokemon-stats/components/Search.tsx:10]: @yung-coder , the Component name should be capitalized.
```suggestion
const Search = (props: SearchProps) => {
```

File: src/plays/pokemon-stats/typing.d.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
+export interface Species {
```
Comment (Sachin-chaurasiya) [inline src/plays/pokemon-stats/typing.d.ts:1]: @yung-coder , Can we change the file name from `typing.d.ts` to `types.d.ts`?

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Requested a few changes.

File: src/plays/pokemon-stats/typing.d.ts (hunk 10 above)
Comment (yung-coder) [reply src/plays/pokemon-stats/typing.d.ts:1]: okay

**Comment by Sachin-chaurasiya** [comment]
@yung-coder , Please do not mark the conversation as resolved before pushing the changes.

File: src/plays/pokemon-stats/PokemonStats.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Sachin-chaurasiya) [reply src/plays/pokemon-stats/PokemonStats.tsx:13]: @yung-coder , Let me know if you need any help, we can connect on discord.

[the remark this answers is not in the document]
Comment (yung-coder) [reply src/plays/pokemon-stats/PokemonStats.tsx:13]: sure

**Comment by yung-coder** [comment]
made the changes @Sachin-chaurasiya  please give it a look

File: src/plays/pokemon-stats/PokemonStats.tsx [hunk 11]
```diff
@@ -6,12 +6,18 @@ import Pokemoncard from "./components/Pokemoncard";
...
+import { PlayRoot, Root } from "./types";
+
 // WARNING: Do not change the entry componenet name
-function PokemonStats(props) {
+function PokemonStats(props: PlayRoot) {
   // Your Code Start below.
-  const [pokemon, setpokemon] = useState();
-  const [name, setname] = useState("");
-  const fetchPokemonDetails = (e) => {
+  const [pokemon, setpokemon] = useState<Root>();
```
Comment (Sachin-chaurasiya) [inline src/plays/pokemon-stats/PokemonStats.tsx:15]:
```suggestion
  const [pokemon, setPokemon] = useState<Root>();
```

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]

**Comment by yung-coder** [comment]
done @Sachin-chaurasiya  please give it a look ....

**Review by Sachin-chaurasiya — APPROVED** [review]

**Comment by yung-coder** [comment]
Hey @atapas can you give it a final review ....

**Comment by koustov** [comment]
I haven't looked at this PR. @Sachin-chaurasiya @atapas if you guys are ok, let's get it merged

**Comment by koustov** [comment]
> I haven't looked at this PR. @Sachin-chaurasiya @atapas if you guys are ok, let's get it merged

Just say @Sachin-chaurasiya's approval. @atapas are you ok with this PR?

**Comment by yung-coder** [comment]
hey @atapas  is it all good to merge...

**Pull request merged**